private/bufpkg/bufmodule/errors.go

```diff
@@ -209,3 +212,47 @@ func (n *NoProtoFilesError) Error() string {
 	_, _ = builder.WriteString(` had no .proto files`)
 	return builder.String()
 }
+
+// DigestMismatchError is the error returned if the Digest of a downloaded Module or Commit
+// does not match the expected digest in a buf.lock file.
+type DigestMismatchError struct {
+	ModuleFullName ModuleFullName
+	CommitID       uuid.UUID
+	ExpectedDigest Digest
+	ActualDigest   Digest
+}
+
+// Error implements the error interface.
+func (m *DigestMismatchError) Error() string {
+	if m == nil {
+		return ""
+	}
+	var builder strings.Builder
+	_, _ = builder.WriteString(`*** Digest verification failed`)
+	if m.ModuleFullName != nil {
+		_, _ = builder.WriteString(` for "`)
+		_, _ = builder.WriteString(m.ModuleFullName.String())
+		if !m.CommitID.IsNil() {
+			_, _ = builder.WriteString(`:`)
+			_, _ = builder.WriteString(uuidutil.ToDashless(m.CommitID))
```

Comment:
Just doing dashless so we don't have to come back to this re: https://github.com/bufbuild/buf/pull/2854